src/java/org/apache/cassandra/db/commitlog/CompressedSegment.java

```diff
@@ -41,15 +48,10 @@ public class CompressedSegment extends FileDirectSegment
     /**
      * Constructs a new segment file.
      */
-    CompressedSegment(CommitLog commitLog, AbstractCommitLogSegmentManager manager)
+    CompressedSegment(AbstractCommitLogSegmentManager manager, ThrowingFunction<Path, FileChannel, IOException> channelFactory)
     {
-        super(commitLog, manager);
-        this.compressor = commitLog.configuration.getCompressor();
-    }
-
-    ByteBuffer createBuffer(CommitLog commitLog)
-    {
-        return manager.getBufferPool().createBuffer();
+        super(manager, channelFactory);
+        this.compressor = manager.commitLog.configuration.getCompressor();
```

Comment:
I hesitate - although I agree that it would look better, I'd like to keep the API minimal. 

OK - actually it seems that exposing commitlog is not something we really need so I'll make the commitlog private in segment manager and expose only `getConfiguration` method as the configuration is mostly immutable.